### ISSUE TITLE
[Bug] Mask trade quest can become incompletable due to Happy Mask Shop collection delay bug

### ISSUE THREAD

**Issue by ETR-BTF** [TestRunnerSRL/OoT-Randomizer#1881]
> Confirming the borrowing of a mask from the Happy Mask Shop and then exiting the shop before receiving the mask will result in the mask trading quest becoming incompletable. The game will set the flag for having borrowed the mask, but it will not be in your inventory. As a result, the owner of the shop will not do business with you until you've sold the mask which you do not possess. In order to be able to leave the shop before receiving the mask, you need to be very quick and confirm the mask while the title card ("Happy Mask Shop") is still up on the screen. This collection delay bug can also occur in other shops, but since other shops don't set flags like the Happy Mask Shop does, the bug is inconsequential elsewhere.

This has been known since roughly 25 sept 2022, but we just never opened an issue. 

With the child trade quest items shuffled the Happy Mask Shop behaves like other shops w.r.t this bug (i.e. the bug is inconsequential).

**Comment by ETR-BTF** [comment]
When trying to replicate this bug it helps to equip the Bunny Hood with the speed-up feature enabled.

**Comment by r0bd0g** [comment]
Duplicate of #1239 which was closed only yesterday. Are you saying it actually hasn't been fixed?

**Comment by ETR-BTF** [comment]
> Duplicate of #1239 which was closed only yesterday. Are you saying it actually hasn't been fixed?

Ah turns out an issue *was* opened I just didn't find it because it was closed.

I specifically tested on mracsys's trade quest rework branch and I was able to reproduce it. I'm not sure why it was closed 2 days ago. Perhaps a misunderstanding about what mracsys's PR included.

Edit: I checked the PR and it was closed because mracsys said it fixed this bug, but it unfortunately still exists.
